x-pack/plugins/ml/public/application/trained_models/models_management/start_deployment_setup.tsx

```diff
@@ -0,0 +1,310 @@
+
+interface StartDeploymentSetup {
+  config: ThreadingParams;
+  onConfigChange: (config: ThreadingParams) => void;
+}
+
+export interface ThreadingParams {
+  numOfAllocations: number;
+  threadsPerAllocations: number;
+}
+
+const THREADS_MAX_EXPONENT = 6;
+
+/**
+ * Form for setting threading params.
+ */
+export const StartDeploymentSetup: FC<StartDeploymentSetup> = ({ config, onConfigChange }) => {
+  const numOfAllocation = config.numOfAllocations;
+  const threadsPerAllocations = config.threadsPerAllocations;
+
+  const threadsPerAllocationsOptions = useMemo(
+    () =>
+      new Array(THREADS_MAX_EXPONENT).fill(null).map((v, i) => {
+        const value = Math.pow(2, i);
+        const id = value.toString();
+
+        return {
+          id,
+          label: id,
+          value,
+        };
+      }),
+    []
+  );
+
+  const toggleIdSelected = threadsPerAllocationsOptions.find(
+    (v) => v.value === threadsPerAllocations
+  )!.id;
+
+  return (
+    <EuiForm component={'form'} id={'startDeploymentForm'}>
+      <EuiDescribedFormGroup
+        title={
+          <h3>
+            <FormattedMessage
+              id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsLabel"
+              defaultMessage="Number of allocations"
+            />
+          </h3>
+        }
+        description={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsHelp"
+            defaultMessage="Increasing the number of allocations will increase throughput, providing your ML nodes have sufficient processors."
+          />
+        }
+      >
+        <EuiFormRow
+          label={
+            <FormattedMessage
+              id="xpack.ml.trainedModels.modelsList.startDeployment.numbersOfAllocationsLabel"
+              defaultMessage="Number of allocations"
+            />
+          }
+          hasChildLabel={false}
+        >
+          <EuiFieldNumber
+            fullWidth
+            min={1}
+            step={1}
+            name={'numOfAllocations'}
+            value={numOfAllocation}
+            onChange={(event) => {
+              onConfigChange({ ...config, numOfAllocations: Number(event.target.value) });
+            }}
+          />
+        </EuiFormRow>
+      </EuiDescribedFormGroup>
+
+      <EuiDescribedFormGroup
+        title={
+          <h3>
+            <FormattedMessage
+              id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationLabel"
+              defaultMessage="Threads per allocation"
+            />
+          </h3>
+        }
+        description={
+          <FormattedMessage
+            id="xpack.ml.trainedModels.modelsList.startDeployment.threadsPerAllocationHelp"
+            defaultMessage="Increasing threads per allocation will decrease latency, providing your ML nodes have sufficient processors."
+          />
+        }
+      >
+        <EuiFormRow
+          label={
```

Comment:
I'm not sure this label is needed as it duplicates the text in the `EuiDescribedFormGroup` title.